pyanaconda/modules/subscription/runtime.py

```diff
@@ -15,14 +15,67 @@
 # License and may only be used or replicated with the express permission of
 # Red Hat, Inc.
 #
+import os
+import json
+
 from dasbus.typing import get_variant, Str
+from dasbus.connection import AddressedMessageBus
+from dasbus.error import DBusError
+
+from pyanaconda.modules.common.constants.objects import RHSM_REGISTER
+
+from pyanaconda.core.i18n import _
 
 from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.constants.services import RHSM
+from pyanaconda.modules.common.constants.objects import RHSM_REGISTER
 
 from pyanaconda.anaconda_loggers import get_module_logger
 log = get_module_logger(__name__)
 
 
+def start_rhsm_private_bus(rhsm_register_server_proxy, message_bus_class=AddressedMessageBus):
```

Comment:
I would consider to create a subclass of `MessageBus` and move this code to the method `_get_connection` (see the implementation of `AnacondaMessageBus`). Also, I would use a context manager instead of these start and stop functions. Ideally, the code in the tasks could look like this:

```
with RHSMPrivateBus() as bus:
    private_register_proxy = bus.get_proxy(...)
    ...

```

Reply:
Sounds good, lets do that. :)

Reply:
Implemented a `RHSMPrivateBus` that inherits `MessageBus` and is also a context manager. The only really difference from the suggestion is that the constructor takes a single argument, which is the RHSM registration server proxy (acquired via the `RHSMObserver` instance).